src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    Animal.alive.forEach((classObj, index, arr) => {
+      if (classObj instanceof Herbivore && classObj.health <= 0) {
+        arr.splice(index, 1);
+      }
+    });
```

Comment:
[CODE PERFORMANCE] - The current implementation of removing dead animals from the Animal.alive array uses a forEach loop with an indexOf check and splice, which can be inefficient and may cause issues when modifying the array during iteration. Consider using the filter method to create a new array with animals that are alive, as it will be more performant and won't have the issues associated with modifying the array during iteration.